addons/mail/static/src/model/model_manager.js

```diff
@@ -216,12 +274,24 @@ class ModelManager {
      */
     deleteAll() {
         this._updateCycle(() => {
-            for (const record of Object.values(this._records)) {
-                record.delete();
-            }
+            this._models.forEach(model => {
+                Object.values(model.records).forEach(record => {
+                    record.delete();
+                });
+            });
         });
     }
 
+    /**
+     * Returns whether the given record still exists.
+     *
+     * @param {mail.model} record
+     * @returns {boolean}
+     */
+    exists(record) {
+        return record.localId in this._models.get(record.constructor).records;
```

Comment:
Would just check truthy value instead of `in`

Reply:
done